### PR TITLE
[#82] TOC Tree Fix and Page Break Handling

### PR DESCRIPTION
**Pull request by kreuzberger**
Pull request also contains the unmerged pull requests required to got my main branch sucessfully working

### CODE REVIEW

File: demo/conf.py [hunk 1]
```diff
@@ -26,6 +26,8 @@
     'sphinxcontrib.plantuml',
     'sphinx_needs',
     'sphinx.ext.autodoc',
+    'sphinx.ext.imgmath',
```
Comment (danwos) [inline demo/conf.py:29]: This PR seems to contain 3 updates:

1. warning filter
2. headline CSS classes
3. math equation 

This line seems to be for 3), but the PR contains no rst-files where math equations are used.
Is this missing?

**Review by danwos — CHANGES_REQUESTED** [review]

File: demo/conf.py (hunk 1 above)
Comment (kreuzberger) [reply demo/conf.py:29]: This was a bugfix for the missing math equastions from #8. The math equations were and are still contained in the demo.rst file of the demo project. The extension to render them was just missing.

**Comment by kreuzberger** [comment]
> Maybe we should split the changes to different PRs, then it is easier to revoke them if one of them makes trouble.
These pull requests are already "in queue" since weeks, but never integrated.
I can break them apart, but handling for me still gets complicated to get a "final" build were all changes are integrated.

**Comment by kreuzberger** [comment]
i divided it into #85. The old ones are still in the queue. I make an additional one for the math stuff

**Pull request closed without merging**